aiohue/v2/controllers/base.py

```diff
@@ -0,0 +1,234 @@
+"""Base controller for HUE resources as retrieved from the Hue bridge."""
+from __future__ import annotations
+
+import asyncio
+from asyncio.coroutines import iscoroutinefunction
+from typing import TYPE_CHECKING, Any, Callable, Dict, Generic, Iterator, List, Tuple
+
+from aiohue.v2.models.connectivity import ZigbeeConnectivity
+from aiohue.v2.models.device import Device
+
+from ...util import dataclass_to_dict, update_dataclass, NoneType
+from ..models.clip import CLIPResource, parse_clip_resource
+from ..models.resource import ResourceTypes
+from .events import EventCallBackType, EventType
+
+if TYPE_CHECKING:
+    from .. import HueBridgeV2
+
+
+EventSubscriptionType = Tuple[
+    EventCallBackType,
+    "Tuple[str] | None",
+    "Tuple[EventType] | None",
+]
+
+
+class BaseResourcesController(Generic[CLIPResource]):
+    """Holds and manages all items for a specific Hue resource type."""
+
+    item_type = ResourceTypes.UNKNOWN
+
+    def __init__(self, bridge: "HueBridgeV2") -> None:
+        """Initialize instance."""
+        self._bridge = bridge
+        self._items: Dict[str, CLIPResource] = {}
+        self._logger = bridge.logger.getChild(self.item_type.value)
+        self._subscribers: List[EventSubscriptionType] = []
+
+    @property
+    def items(self) -> List[CLIPResource]:
+        """Return all items for this resource."""
+        return list(self._items.values())
+
+    async def initialize(self, initial_data: List[dict] | None) -> None:
+        """
+        Initialize controller by fetching all items for this resource type from bridge.
+
+        Must be called only once. Any updates will be retrieved by events.
+        Optionally provide the initial data.
+        """
+        if initial_data is None:
+            endpoint = f"clip/v2/resource/{self.item_type.value}"
+            initial_data = await self._bridge.request("get", endpoint)
+        item_count = 0
+        for item in initial_data:
+            if ResourceTypes(item["type"]) != self.item_type:
+                continue
+            item_count += 1
+            res: CLIPResource = parse_clip_resource(item)
+            self._items[res.id] = parse_clip_resource(item)
```

Comment:
You're not using `res`

```suggestion
            self._items[res.id] = parse_clip_resource(item)
```